app/js/actions/labOrders/fetchLabOrders.js

```diff
@@ -7,30 +7,15 @@ import {
   FETCH_LAB_ORDERS_LOADING,
 } from '../actionTypes';
 
-const fetchLabOrdersSuccess = orders => ({
-  orders,
-  type: FETCH_LAB_ORDERS_SUCCESS,
-});
-
-const fetchLabOrdersFailure = error => ({
-  error,
-  type: FETCH_LAB_ORDERS_FAILURE,
-});
 
-const fetchLabOrdersLoading = status => ({
-  status,
-  type: FETCH_LAB_ORDERS_LOADING,
+const fetchLabOrders = (
+  uri,
+  limit = 5,
+  patient,
+) => ({
+  type: 'FETCH_LAB_ORDERS',
```

Comment:
We don't need to handle `SUCCESS`, `FAILURE` or `LOADING` cos the middleware does that for us. We just return a thunk as an async action and since it returns a promise, it'll map directly to whatever value it returns ie. pending, resolved or rejected

Reply:
Good job @rhenshaw56 
This is such a nice refactoring. With `redux-promise-middleware`, we can reduce the amount of codes we write and it would even make the codebase better to understand.

But there is something I am still confused about with the explanation you gave above: 
My understading on how to use the `redux-promise-middleware` is to return an action(as an object), but the action can have a promise payload. The explanation above states that _`...We just return a thunk... `_, I am not sure of what you are referring to with this since you are not even returning a `thunk` in the code you wrote. Can you please provide clarity on that?

Reply:
I see that you only made this refactoring(using `redux-promise-middleware` pattern) in `fetchLabOrdersAction.js` and `labConceptsAction.js`, do you plan on refactoring other action files to also follow this new pattern?

Reply:
yes @IAMOTZ I plan to do that! currently doing that but for each file I touch, the actions and reducers tests break, code breaks in the UI also based on maybe how it was imported and how the reducer read the data, so it's gonna be a bumpy ride.

Reply:
@mogoodrich yes it will generate those actions automatically depending on the state of the promise (pending, fulfilled, failed), using sagas combined with redux-promise-middleware is on it's own very powerful in the redux arsenal, cos with sagas, it's easier to trigger optimistic updates and the promise middleware makes it easier to reduce logic for fetching data which means easier debugging and reduction in code size, reducing the number of problems one could have lol! yeah, we could add it to be used alongside the sagas